eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java

```diff
@@ -353,6 +354,47 @@ void choosesTransformerFactoryInConcurrentEnvironment() {
         }
     }
 
+    @Test
+    void failsOnErrorAfterNotReplacingWrongVersions(@TempDir final Path tmp) {
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> new FakeMaven(tmp)
+                .withProgram(
+                    "+package f\n",
+                    "[] > main",
+                    "  seq|99.99.99 > @",
+                    "    nop"
+                )
+                .with("withVersions", true)
+                .with("hashes", new CommitHashesMap.Fake())
+                .execute(new FakeMaven.Optimize()),
+            "Program should have failed on error on optimization step with wrong tag, but it didn't"
+        );
+    }
+
+    @Test
+    void containsValidReplacedVersionAfterReplacingAndOptimization(@TempDir final Path tmp)
+        throws Exception {
+        new FakeMaven(tmp)
+            .withProgram(
+                "+package f\n",
+                "[] > main",
+                "  seq|0.28.10 > @",
+                "    nop"
+            )
+            .with("withVersions", true)
+            .with("hashes", new CommitHashesMap.Fake())
+            .execute(new FakeMaven.Optimize());
+        MatcherAssert.assertThat(
```

Comment:
@maxonfjvipon Please, add a human-readable description for that assertion:
https://www.yegor256.com/2023/01/19/layout-of-tests.html#assertions

Reply:
@maxonfjvipon Also, instead of "Matchers.hasSize(1)" you can use `Matchers.not(Matchers.empty())`

Reply:
But it's optional